sympy/polys/matrices/sdm.py

```diff
@@ -28,6 +28,30 @@ def __init__(self, elemsdict, shape, domain):
         if not all(0 <= c < n for row in self.values() for c in row):
             raise DDMBadInputError("Column out of range")
 
+    def getitem(self, i, j):
+        m, n = self.shape
+        if i >= m:
+                raise IndexError("row index out of range")
+        if j >= n:
+            raise IndexError("column index out of range")
+
+        try:
+            return self[i][j]
+        except KeyError:
+            return self.domain.zero
+
+    def extract_slice(self, slice1, slice2):
+        m, n = self.shape
+        ri = range(m)[slice1]
+        ci = range(n)[slice2]
+
+        sdm = {}
+        for i, row in self.items():
+            if i in ri:
+                sdm[ri.index(i)] = {ci.index(j): e for j, e in row.items() if j in ci}
```

Comment:
As previously mentioned if the dict on the rhs is empty then it shouldn't be added. Only nonzero rows are stored in SDM.

Reply:
Sure, forgot by mistake.
Thanks.